dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/context/annotation/DubboClassPathBeanDefinitionScanner.java

```diff
@@ -67,4 +74,35 @@ public boolean checkCandidate(String beanName, BeanDefinition beanDefinition) th
         return super.checkCandidate(beanName, beanDefinition);
     }
 
+    @Override
+    public Set<BeanDefinition> findCandidateComponents(String basePackage) {
+        Set<BeanDefinition> beanDefinitions = beanDefinitionMap.get(basePackage);
```

Comment:
No need to cache BeanDefinition, use the result of the first scan directly outside